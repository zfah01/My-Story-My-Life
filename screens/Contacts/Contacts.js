import React, { useState, useEffect } from 'react';
import { ImageBackground, SafeAreaView, Button, TouchableOpacity, Platform, ScrollView, StyleSheet, Text, TextInput, View, Image, Alert, Linking } from 'react-native';
import * as MailComposer from 'expo-mail-composer';
import Help from '../Help/Help';

// Imports the documents styling.

// Imports firestore from firebase to save user entries to the firstore database.
//import firestore from '@react-native-firebase/firestore';
import { db, auth } from '../../firebase/firebase';
import { Link } from '@react-navigation/native';

export default function Contacts(props) {
    const [isAvailable, setIsAvailable] = useState(false);
    const [recipients, setRecipients] = useState([]);
    const [subject, setSubject] = useState(undefined);
    const [body, setBody] = useState(undefined);
    const [email, setEmail] = useState(undefined);

    const [helpPressed, setHelpPressed] = useState(false);

    const contactsRef = db.collection('Contacts');
    // Gets the users ID from props passed in from App.js.
    const userID = props.extraData.id;
    const user = auth.currentUser;

    const closeHelp = () => {
      setHelpPressed(false);
  };

    useEffect(() =>{
        async function checkAvailability(){
            const isMailAvailable = await MailComposer.isAvailableAsync();
            setIsAvailable(isMailAvailable);
        }
        checkAvailability();
    }, []);

    const sendMail = ()=> {
      
      if(recipients.length !== 0){
        MailComposer.composeAsync({
          subject: subject,
          body: body,
          recipients: recipients
        });
       
      } else {
        Alert.alert("Please enter a recipient");
      }




      setSubject(undefined);
      setBody(undefined);
      setRecipients([]);

    };

    

     // Checks if email is valid format.
  const checkEmail = (str) => {

  const validEmail = /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;

  return validEmail.test(str);
};

    const addRecipient= ()=> {
      let newRecipients = [...recipients];
      if (!checkEmail(email)) {
        alert('Please ensure email is valid.');
        return;
     }

      newRecipients.push(email);

      setRecipients(newRecipients);
      setEmail(undefined);
    }

    const showRecipients = ()=> {
      if(recipients.length === 0){
        return <Text> No recipients added</Text>
      }
      
      return recipients.map((recipient, index)=>{
        return <Text key={index}>{recipient}</Text>;
      })
    }
    if (helpPressed == false) {
    return(
        <SafeAreaView style={styles.topView}>
            <Text style={styles.topSent}>Stay in touch with your Carer, Family or Friends</Text>
            <Text style={styles.bottomSent}>Rediscover your lost memories or get any help</Text>
            <View style={styles.contentContainer}>
              {showRecipients()}
              <View style={styles.addRec}>
            <TextInput value={email} onChangeText={setEmail} placeholder="Email"/>
            <Button title="Add Recipient" onPress={addRecipient}/>
            </View>
            
            </View>
            <View style={styles.contentContainer}>
            <TextInput value={subject} onChangeText={setSubject} placeholder="Subject"/>
           
            <View style={styles.bod}>
            <TextInput value={body} multiline={true} style={styles.bodText} onChangeText={setBody} placeholder="Body"/>
             </View>
             {isAvailable ? <Button title='Send Mail' onPress={sendMail}/> : <Text>Email not available</Text>}
        </View>

        <TouchableOpacity style={styles.GoProBox} onPress={() => setHelpPressed(true)}>
            <ImageBackground source={require('../../assets/faqs-customer-service-icon-concept.jpg')} resizeMode='cover' style={styles.goProBgImage}>
            <View style={styles.overlayView}/>
            <Text style={styles.goProText}> Help and Advice</Text>
            </ImageBackground>
            </TouchableOpacity>
        </SafeAreaView>
    );
  }else {
    return (
        <Help {...props}  extraData={userID} closeHelp={closeHelp} />
    );
  }
   
}

const styles = StyleSheet.create({
  contentContainer: {
    margin: 35,
    padding: Platform.OS === 'ios'? 10: 5,
    backgroundColor: '#FFFFFF',
    borderRadius: 25,
    marginBottom: Platform.OS === 'ios'? null: 15,
},


  topSent: {
    fontSize : 20,
    textAlign : 'center',
    fontWeight : 'bold',
    paddingBottom: 10
  },
  bottomSent: {
    fontSize : 15,
    textAlign : 'center',
    fontStyle: 'italic',
 
  }, 
  bod: {
    alignItems: 'center',
  },
  addRec: {
    alignItems: 'flex-end',
  },
  bodText: {
    flexWrap: 'wrap',
    height: 80,
    margin: 20,
    padding: 10,

  },
 
  titleH: {
    textAlign: 'center',
    fontWeight: 'bold',
    paddingBottom: 10,
    fontSize: 18
  },
  links :{
    color: 'red',
    paddingBottom: 2,
    fontSize: 15

  },
  topView: {
    paddingTop: Platform.OS === 'ios'? 80: 7,
    backgroundColor: '#AFEEEE',
    flex: 1, 
    
  },
  GoProBox: {
    width: '85%',
    height: 200,
    margin: 35,
    backgroundColor: '#00cc00',
    borderRadius: 25,
    alignSelf: 'center',
    overflow: 'hidden'

},
goProBgImage: {
    width: '100%', height: '100%',


},

goProText: {
    textAlign: 'center',
    fontSize: 20,
    marginTop: 140,
    fontWeight: 'bold',
    padding: 10,
    color: 'white'

},

overlayView: {
    height: "100%",
    width: "100%",
    position: 'absolute',
    backgroundColor: 'rgba(0, 204, 0, 0.5)',

}


});