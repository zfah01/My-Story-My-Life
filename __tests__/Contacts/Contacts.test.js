import 'react-native';
import React from 'react';
import Contacts from '../../screens/Contacts/Contacts';

import { cleanup, render } from '@testing-library/react-native';

afterEach(cleanup);

describe('Test to check Contacts page is rendered and if user can press help and advice', () => {
  const extraData = { 'id': 1, 'name': 'Stephen James' };
  const contactsRender = render(<Contacts extraData={extraData} />).toJSON();
  
  it('Checks to see if the Contacts page renders', () => {
      expect(contactsRender).toMatchSnapshot();
  });

});