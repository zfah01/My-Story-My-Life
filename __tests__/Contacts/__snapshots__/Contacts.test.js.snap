// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test to check Contacts page is rendered and if user can press help and advice Checks to see if the Contacts page renders 1`] = `
<RCTSafeAreaView
  style={
    Object {
      "backgroundColor": "#AFEEEE",
      "flex": 1,
      "paddingTop": 80,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 20,
        "fontWeight": "bold",
        "paddingBottom": 10,
        "textAlign": "center",
      }
    }
  >
    Stay in touch with your Carer, Family or Friends
  </Text>
  <Text
    style={
      Object {
        "fontSize": 15,
        "fontStyle": "italic",
        "textAlign": "center",
      }
    }
  >
    Rediscover your lost memories or get any help
  </Text>
  <View
    style={
      Object {
        "backgroundColor": "#FFFFFF",
        "borderRadius": 25,
        "margin": 35,
        "marginBottom": null,
        "padding": 10,
      }
    }
  >
    <Text>
       No recipients added
    </Text>
    <View
      style={
        Object {
          "alignItems": "flex-end",
        }
      }
    >
      <TextInput
        onChangeText={[Function]}
        placeholder="Email"
      />
      <View
        accessibilityRole="button"
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "margin": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            Add Recipient
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#FFFFFF",
        "borderRadius": 25,
        "margin": 35,
        "marginBottom": null,
        "padding": 10,
      }
    }
  >
    <TextInput
      onChangeText={[Function]}
      placeholder="Subject"
    />
    <View
      style={
        Object {
          "alignItems": "center",
        }
      }
    >
      <TextInput
        multiline={true}
        onChangeText={[Function]}
        placeholder="Body"
        style={
          Object {
            "flexWrap": "wrap",
            "height": 80,
            "margin": 20,
            "padding": 10,
          }
        }
      />
    </View>
    <Text>
      Email not available
    </Text>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "#00cc00",
        "borderRadius": 25,
        "height": 200,
        "margin": 35,
        "opacity": 1,
        "overflow": "hidden",
        "width": "85%",
      }
    }
  >
    <View
      accessibilityIgnoresInvertColors={true}
      style={
        Object {
          "height": "100%",
          "width": "100%",
        }
      }
    >
      <Image
        resizeMode="cover"
        source={1}
        style={
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "height": "100%",
              "width": "100%",
            },
            undefined,
          ]
        }
      />
      <View
        style={
          Object {
            "backgroundColor": "rgba(0, 204, 0, 0.5)",
            "height": "100%",
            "position": "absolute",
            "width": "100%",
          }
        }
      />
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 20,
            "fontWeight": "bold",
            "marginTop": 140,
            "padding": 10,
            "textAlign": "center",
          }
        }
      >
         Help and Advice
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;
